### PR TITLE
QuadMesh remove format ARRAY_FORMAT_INDEX

### LINKED ISSUE #53133: [4.0] QuadMesh generate_triangle_mesh returns null
### Godot version

4.0.dev (73668c59aa)

### System information

Windows 10

### Issue description

The following script does not produce a triangle mesh. Instead, `generate_triangle_mesh` returns null.
```gdscript
var mesh := QuadMesh.new()
var tris = mesh.generate_triangle_mesh()
```

This happens because all PrimitiveMesh types are formatted to include indices.
https://github.com/godotengine/godot/blob/98a7bf340698d89708aae7c26f0e1416bd6cdc58/scene/resources/primitive_meshes.cpp#L151

When generating the triangle mesh, it checks if it is formatted to include indices.
https://github.com/godotengine/godot/blob/98a7bf340698d89708aae7c26f0e1416bd6cdc58/scene/resources/mesh.cpp#L56

However, QuadMesh does not set indices.
https://github.com/godotengine/godot/blob/98a7bf340698d89708aae7c26f0e1416bd6cdc58/scene/resources/primitive_meshes.cpp#L1377

### Steps to reproduce

See script above

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by briansemrau**
Fixes #53133

### CODE REVIEW

File: scene/resources/primitive_meshes.cpp [hunk 1]
```diff
@@ -1390,6 +1390,13 @@ void QuadMesh::_bind_methods() {
 	ADD_PROPERTY(PropertyInfo(Variant::VECTOR3, "center_offset"), "set_center_offset", "get_center_offset");
 }
 
+uint32_t QuadMesh::surface_get_format(int p_idx) const {
+	ERR_FAIL_INDEX_V(p_idx, 1, 0);
+
+	return RS::ARRAY_FORMAT_VERTEX | RS::ARRAY_FORMAT_NORMAL | RS::ARRAY_FORMAT_TANGENT | RS::ARRAY_FORMAT_TEX_UV;
+}
+
+
```
Comment (clayjohn) [inline scene/resources/primitive_meshes.cpp:1399]: This extra blank line is causing CI to fail. Just needs to be deleted

Comment (briansemrau) [reply scene/resources/primitive_meshes.cpp:1399]: Looks like I'll never have a clean PR without a few force pushes

Comment (clayjohn) [reply scene/resources/primitive_meshes.cpp]: Haha! I don't know when the last time I made a PR without force pushing was.

**Review by clayjohn — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**